src/run/emulator.rs

```diff
@@ -1654,6 +1697,88 @@ impl Machine {
                             Err(ExecutionError::new("invalid operand to EcPairing instruction", &self.state, None))
                         }
 
+                    }
+                    Opcode::AVMOpcode(AVMOpcode::NewBuffer) => {
+                        let _size = self.stack.pop_uint(&self.state)?;
+                        self.stack.push(Value::new_buffer(vec![0; 256]));
+                        self.incr_pc();
+                        Ok(true)
+                    }
+                    Opcode::AVMOpcode(AVMOpcode::GetBuffer8) => {
+                        let buf = self.stack.pop_buffer(&self.state)?;
+                        let offset = self.stack.pop_usize(&self.state)?;
```

Comment:
It's probably better to put the args on the stack in the opposite order, with the offset on top.  That's more efficient because the offset is more likely to be a constant, which can be provided as an immediate value to the GetBuffer8 instruction.

(Same applies to GetBuffer64 and GetBuffer 256.)